### ISSUE TITLE
Data option to process a template file but not write it to the output directory

### ISSUE THREAD

**Issue by zachleat** [11ty/eleventy#61]
---
write: false
---

This would be similar to the `_underscoreprefix.html` in jekyll.

**Comment by chrisdmacrae** [comment]
I think `output: false|true` would make the most sense.

It would also be create for the json files that map to entire folder(s).

**Comment by zachleat** [comment]
Related: #188

**Comment by zachleat** [comment]
Note that this matter option could also be set in template / directory data files.

**Comment by adamsilver** [comment]
Would be good to be able to have draft articles sent to the output with a flag:

```npx eleventy --drafts```

**Comment by zachleat** [comment]
A working implementation of this is now in the `permalink-false` branch for marinating/testing/review

**Comment by edwardhorsford** [comment]
@zachleat does having `permalink: false` allow conditional output depending on environment variables?

Eg as per @adamsilver above - being able to set `--drafts` to render those pages marked draft. Or not render them if the arg is not included? Those draft pages will have permalinks assigned for testing, it's just that I want to be able to *not* render them in production.

My reading of it currently is that by reusing `permalink` for this, it means pages can't have a permalink applied. Perhaps I'm misunderstanding the feature though.

**Comment by zachleat** [comment]
@edwardhorsford yes, for data properties you can set them in data files (JS or JSON) or in front matter. This data cascade is documented here: https://www.11ty.io/docs/data/

With this feature, @adamsilver should be able to implement this himself by creating a directory-specific JS data file that looks for a `--drafts` command line flag. https://www.11ty.io/docs/data-js/ It could be conditional based on development or production environment variables too, sure.

E.g. a `drafts.11tydata.js` file in a `drafts/` folder, continuing from this example: https://www.11ty.io/docs/data-js/#example%3A-exposing-environment-variables

```js
module.exports = {
  permalink: process.env.ELEVENTY_ENV === "production" ? false : "/{{ fileSlug }}/"
};
```

**Comment by edwardhorsford** [comment]
@zachleat if an individual page already has a permalink, would that still work?

In my head, a page would have both a permalink *AND* a drafts flag. And something separate evaluates the both to decide whether to process.

**Comment by edwardhorsford** [comment]
@zachleat unless I'm misunderstanding, I think this could be tricky you don't keep drafts all in one folder. Each time you want to have a conditional draft, it's a lot of extra code to add.

Turning a draft on and off isn't a case of adding `status: draft` or similar to frontmatter (unless I've got it wrong).

Idea - would something like this work in eleventy config?
```js
if (process.env.NODE_ENV == production){
	eleventyConfig.ignoreFiles("status", "draft");
}
```

**Comment by zachleat** [comment]
Hm? If you wanted to do this for an individual file you could use a template specific JS data file or probably even in front matter if you use the JS front matter type: https://www.11ty.io/docs/data-frontmatter/#javascript-front-matter

However, I’d rather move bikeshed discussions for a higher level drafts feature to https://github.com/11ty/eleventy/issues/188 if we could!

**Comment by zachleat** [comment]
Related: #244, #245.
